src/contrib/inference/dynamichmc.jl

```diff
@@ -0,0 +1,65 @@
+###
+### DynamicHMC backend - https://github.com/tpapp/DynamicHMC.jl
+###
+struct DynamicNUTS{AD, space} <: Hamiltonian{AD} end
+
+"""
+    DynamicNUTS()
+
+Dynamic No U-Turn Sampling algorithm provided by the DynamicHMC package.
+To use it, make sure you have the DynamicHMC package installed.
+
+"""
+DynamicNUTS(args...) = DynamicNUTS{ADBackend()}(args...)
+DynamicNUTS{AD}() where AD = DynamicNUTS{AD, ()}()
+function DynamicNUTS{AD}(space::Symbol...) where AD
+    DynamicNUTS{AD, space}()
+end
+
+getspace(::Type{<:DynamicNUTS{<:Any, space}}) where {space} = space
+getspace(alg::DynamicNUTS{<:Any, space}) where {space} = space
+
+function sample(
+    rng::AbstractRNG,
+    model::Model,
+    spl::Sampler{<:DynamicNUTS},
+    N::Integer;
+    kwargs...
+)
+    samples = Array{Transition}(undef, N)
+
+    runmodel!(model, spl.state.vi, SampleFromUniform())
+
+    if spl.selector.tag == :default
```

Comment:
This line assumes that we are using HMC (DynamicNUTS) for all model parameters when the condition `spl.selector.tag == :default` is true. This is not very clear from the code itself. Maybe 

- add a comment here
- or, move this block into the `sample_init!` function.

Reply:
Is this a bug? Shouldn't this just be using `SampleFromUniform` but with all the parameters transformed?

Reply:
Sorry for the confusion. The code is correct here. What I suggested is a bit refactoring, since this code appears in all HMC related samplers. It might make sense to create a function and call the function in all places.

Reply:
It's in the `sample_init!` function now. Can we close this?